server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -313,6 +319,22 @@ protected void sendProxyRequest(
     // will log that on the remote node.
     clientRequest.setAttribute(AuthConfig.DRUID_AUTHORIZATION_CHECKED, true);
 
+    // Check if there is an authentication result and use it to decorate the proxy request if needed.
+    AuthenticationResult authenticationResult = (AuthenticationResult) clientRequest.getAttribute(
+        AuthConfig.DRUID_AUTHENTICATION_RESULT);
+    if (authenticationResult != null && authenticationResult.getAuthenticatedBy() != null) {
+      Authenticator authenticator = authenticatorMapper.getAuthenticatorMap()
+                                                       .get(authenticationResult.getAuthenticatedBy());
+      if (authenticator != null) {
```

Comment:
If I am not wrong user needs to make sure that he defines a property like this for every authenticator -  
druid.auth.authenticator.<MyAuthenticatorName>.name=<MyAuthenticatorName> 
If it possible to infer the name without making the user add this property ? 
If not lets make sure we document this.

Reply:
it is coming from the list of authenticators and injected at the Authenticator module.

Reply:
sounds good.